R/geom_highlight.R

```diff
@@ -165,32 +184,35 @@ bleach_layer <- function(layer, group_key,
   # remove colour and fill from mapping
   layer$mapping[c("colour", "fill")] <- list(NULL)
 
-  if (!is.null(group_key)) {
+  if (!is.null(group_info$key)) {
     # Add group var to preserver implicit grouping to prevent the bleached
     # data to facetted, rename the group column to the very improbable name.
     # e.g. geom_line(aes(colour = c)); if colour aes is removed, line will be drawn unintentionally.
     # But, is group key always needed...? (e.g. points)
-    layer$data <- dplyr::rename(layer$data, !!VERY_SECRET_COLUMN_NAME := !!group_key)
+    secret_colname <- rlang::expr_text(VERY_SECRET_COLUMN_NAME)
+    names(group_info$key) <- paste0(secret_colname, seq_along(group_info$key))
+    layer$data <- dplyr::rename(layer$data, !!!group_info$key)
```

Comment:
It's not always that group keys can be renamed; it may be mapped from the other aes.
Usually, this won't be the problem because it is typically `colour` or `fill`, which are overwritten by grey.
But, if `x` is used for grouping, `x` is necessary. Other aes like `shape` might be also the case.